test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SqlAssessment/SqlAssessmentServiceTests.cs

```diff
@@ -0,0 +1,170 @@
+                        Assert.False(string.IsNullOrWhiteSpace(i.Message), "Message is null, empty, or white space");
+
+                        if (i.Kind == 0)
+                        {
+                            AssertInfoPresent(i);
+                        }
+                    });
+        }
+
+        private static async Task<AssessmentResult<TResult>> CallAssessment<TResult>(
+            string methodName,
+            SqlObjectType sqlObjectType,
+            string databaseName = "master")
+            where TResult : AssessmentItemInfo
+        {
+            var liveConnection = LiveConnectionHelper.InitLiveConnectionInfo(databaseName);
+            var connInfo = liveConnection.ConnectionInfo;
+
+            AssessmentResult<TResult> response;
+
+            using(var service = new SqlAssessmentService(
+                TestServiceProvider.Instance.ConnectionService,
+                TestServiceProvider.Instance.WorkspaceService))
+            {
+                string randomUri = Guid.NewGuid().ToString();
+                AssessmentParams requestParams =
+                    new AssessmentParams { OwnerUri = randomUri, TargetType = sqlObjectType };
+                ConnectParams connectParams = new ConnectParams
+                                                  {
+                                                      OwnerUri = requestParams.OwnerUri,
+                                                      Connection = connInfo.ConnectionDetails,
+                                                      Type = ConnectionType.Default
+                                                  };
+
+                var methodInfo = typeof(SqlAssessmentService).GetMethod(
+                    methodName,
+                    BindingFlags.Instance | BindingFlags.NonPublic);
+
+                Assert.NotNull(methodInfo);
+
+                var func = (AssessmentMethod<TResult>)Delegate.CreateDelegate(
+                    typeof(AssessmentMethod<TResult>),
+                    service,
+                    methodInfo);
+
+                response = await service.CallAssessmentEngine<TResult>(
+                               requestParams,
+                               connectParams,
+                               randomUri,
+                               t => func(t));
+            }
+
+            Assert.NotNull(response);
+            if (response.Success)
+            {
+                Assert.All(
+                    response.Results,
+                    i =>
+                        {
+                            Assert.Equal(sqlObjectType, (SqlObjectType)i.TargetType);
+                            Assert.Contains(i.Level, AllowedSeverityLevels);
+                        });
+            }
+
+            return response;
+        }
+
+        private void AssertInfoPresent(AssessmentItemInfo item)
+        {
+
+            Assert.False(string.IsNullOrWhiteSpace(item.CheckId), nameof(AssessmentItemInfo.CheckId));

```

Comment:
Suggest adding a message to these asserts so if they fail it's clear what the issue is (since the default message doesn't usually say what values it was checking)